action.yml

```diff
@@ -0,0 +1,59 @@
+name: Project Checks
+description: Project checks commonly used across containerd org
+inputs:
+  working-directory:
+    required: true
+    description: Go project path to run checks in
+runs:
+  using: composite
+  steps:
+    - name: Set env
+      shell: bash
+      run: |
+        echo "GOPATH=${{ github.workspace }}" >> $GITHUB_ENV
+        echo "${{ github.workspace }}/bin" >> $GITHUB_PATH
+
+    - name: Install dependencies
+      shell: bash
+      env:
+        GO111MODULE: off
+      run: |
+        echo "::group::🚧 Get dependencies"
+        go get -u -v github.com/vbatts/git-validation
+        go get -u -v github.com/kunalkushwaha/ltag
+        go get -u -v github.com/LK4D4/vndr
+        echo "::endgroup::"
+
+    - name: DCO Checks
+      shell: bash
+      working-directory: ${{ inputs.working-directory }}
+      env:
+        GITHUB_COMMIT_URL: ${{ github.event.pull_request.commits_url }}
+        DCO_VERBOSITY: "-q"
+        DCO_RANGE: ""
+      run: |
+        echo "::group::👮 DCO checks"
+        set -x
+        if [ -z "${GITHUB_COMMIT_URL}" ]; then
+        DCO_RANGE=$(jq -r '.after + "..HEAD"' ${GITHUB_EVENT_PATH})
+        else
+        DCO_RANGE=$(curl ${GITHUB_COMMIT_URL} | jq -r '.[0].parents[0].sha + "..HEAD"')
+        fi
+        ${{ github.action_path }}/script/validate/dco
+        echo "::endgroup::"
+
+    - name: Validate file headers
+      shell: bash
+      working-directory: ${{ inputs.working-directory }}
+      run: |
+        echo "::group::📚 File headers"
+        ${{ github.action_path }}/script/validate/fileheader ${{ github.action_path }}/
```

Comment:
maybe expose another input variable for the --excludes options so it's more obvious to the uses of this what is being excluded on the header check, see fileheader:
```
ltag -t "${BASEPATH}script/validate/template" --excludes "vendor contrib" --check -v
```

Reply:
i believe ltag omits some directories like vendor by default

Reply:
This one is easy to break with existing CI, let do it separately once we migrated.

Reply:
fair..